wordpress-mobile/WordPress-Android
# Experimental per-app language preferences

This PR adds support for AndroidX [per-app language preferences](https://developer.android.com/guide/topics/resources/app-languages). Since this is a high-risk feature I've hidden it behind a feature flag, with the hope that if all goes well a subsequent PR will entirely drop our custom (buggy!) in-app locale handling and remove `LocaleAwareActivity`.

Note that per-app language prefs are backwards compatible thanks to AndroidX, but the ability to choose the language in the system settings is only available on devices running Android 13 (API 33) or later. For this reason, on pre-13 devices we still have to rely on our custom language picker.

Test 1:

* Run the app on a device using Android 13 or later
* Enable "Experimental per-app language preferences" in Experimental Features
* Go to App Settings and tap "Interface language"
* Note that the system dialog for "App Language" appears instead of our own
* Verify changing languages works

Test 2:

* Perform the above with an older device and verify our own language dialog appears and changing languages works

Test 3:

* Run the app on a device using Android 13  or later
* Disable the feature flag
* Change the app language using our own language picker
* Enable the feature flag
* Verify the previously selected language is honored
